libp2p/network/stream/net_stream.py

```diff
@@ -51,14 +51,14 @@ def set_protocol(self, protocol_id: TProtocol) -> None:
         except MuxedStreamReset as error:
             raise StreamReset() from error
 
-    async def write(self, data: bytes) -> int:
+    async def write(self, data: bytes) -> None:
```

Comment:
what happens if we can't write all of `data` to the underlying stream? the idea w/ this interface is to allow a caller to detect this scenario and re-attempt the write if that happens

Reply:
Good point. I'm changing this because I didn't see this use case(returns before all of the data is written) present in our codebase and also the underlying transport e.g. [`trio.SocketStream.send_all`](https://trio.readthedocs.io/en/stable/reference-io.html#trio.abc.SendStream.send_all). `write` is asynchronous and we probably can guarantee that `write` returns only when all of the data is written. I can also change it back if we would prefer to allow `write` to return if it only writes part of the data.

Reply:
we should avoid underlying implementations from guiding our interface choices but i suppose we can leave this alone for now just to bias towards speed :)